x-pack/plugins/spaces/public/share_saved_objects_to_space/share_saved_objects_to_space_action.tsx

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import React from 'react';
+import { i18n } from '@kbn/i18n';
+import { NotificationsStart } from 'src/core/public';
+import {
+  SavedObjectsManagementAction,
+  SavedObjectsManagementRecord,
+} from '../../../../../src/plugins/saved_objects_management/public';
+import { ShareSavedObjectsToSpaceFlyout } from './components';
+import { SpacesManager } from '../spaces_manager';
+
+export class ShareToSpaceSavedObjectsManagementAction extends SavedObjectsManagementAction {
+  public id: string = 'share_saved_objects_to_space';
+
+  public euiAction = {
+    name: i18n.translate('xpack.spaces.management.shareToSpace.actionTitle', {
+      defaultMessage: 'Share to space',
+    }),
+    description: i18n.translate('xpack.spaces.management.shareToSpace.actionDescription', {
+      defaultMessage: 'Share this saved object to one or more spaces',
+    }),
+    icon: 'share',
+    type: 'icon',
+    enabled: (object: SavedObjectsManagementRecord) => {
```

Comment:
What do you think about hiding the action altogether, rather than enabling/disabling? The disabled option has the potential to confuse users, since there's nothing they can to do ever enable this, short of submitting a PR to make that saved object type sharable. If we choose to keep it visible but disabled, I think it'd be worthwhile including a tooltip explaining why the action isn't available

Reply:
I'd lean towards the tooltip approach. It seemed off to me that the option would just not exist for some object types. I think this conveys that the object is not shareable. Interested to hear if @mdefazio has any thoughts on the matter!

Reply:
Update: tooltip isn't feasible, as we can only pass in a `name` string and `icon` string. Can't pass in a custom element.

I'll change the Share action so it doesn't show up for non-multi-namespace objects, and I'll change the Copy action so it doesn't show up for namespace-agnostic objects.